example/tutorial_dorefanet_mnist_cnn.py

```diff
@@ -0,0 +1,108 @@
+#! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+import time
+import tensorflow as tf
+import tensorlayer as tl
+
+X_train, y_train, X_val, y_val, X_test, y_test = \
+                tl.files.load_mnist_dataset(shape=(-1, 28, 28, 1))
+# X_train, y_train, X_test, y_test = tl.files.load_cropped_svhn(include_extra=False)
+
+sess = tf.InteractiveSession()
+
+batch_size = 128
+
+x = tf.placeholder(tf.float32, shape=[batch_size, 28, 28, 1])
+y_ = tf.placeholder(tf.int64, shape=[batch_size])
+
+
+def model(x, is_train=True, reuse=False):
+    # In BNN, all the layers inputs are binary, with the exception of the first layer.
+    # ref: https://github.com/itayhubara/BinaryNet.tf/blob/master/models/BNN_cifar10.py
+    with tf.variable_scope("binarynet", reuse=reuse):
+        net = tl.layers.InputLayer(x, name='input')
+        net = tl.layers.DorefaConv2d(net, 32, (5, 5), (1, 1), padding='SAME', b_init=None, name='bcnn1')
```

Comment:
How does `DorefaConv2d` know that `32` is for `n_filter` but not for `bitW` or `bitA`?

Reply:
it crashed with
```
$ python3 example/tutorial_dorefanet_mnist_cnn.py 
[TL] Load or Download MNIST > data/mnist
[TL] data/mnist/train-images-idx3-ubyte.gz
[TL] data/mnist/t10k-images-idx3-ubyte.gz
2018-03-22 21:30:07.666529: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2
[TL] InputLayer  binarynet/input: (128, 28, 28, 1)
Traceback (most recent call last):
  File "example/tutorial_dorefanet_mnist_cnn.py", line 48, in <module>
    net_train = model(x, is_train=True, reuse=False)
  File "example/tutorial_dorefanet_mnist_cnn.py", line 25, in model
    net = tl.layers.DorefaConv2d(net, 32, (5, 5), (1, 1), padding='SAME', b_init=None, name='bcnn1')
  File "/home/vagrant/.local/lib/python3.6/site-packages/tensorlayer/layers/binary.py", line 687, in __init__
    act.__name__))
TypeError: %d format: a number is required, not tuple
```

Reply:
I forgot it and I have fixed the bug